### PR TITLE
Make sure RHSM HTTP proxy port from kickstart is an integer

### PR DESCRIPTION
**Pull request by M4rtinK**
Make sure RHSM HTTP proxy port from kickstart is always an integer
before being passed to set_server_proxy(). Also make sure -1 is
passed if proxy port is unknown.

Resolves: rhbz#1794025

### CODE REVIEW

File: pyanaconda/modules/subscription/subscription.py [hunk 1]
```diff
@@ -229,8 +229,13 @@ def process_kickstart(self, data):
             try:
                 proxy = util.ProxyString(data.rhsm.proxy)
                 if proxy.host:
+                    # ensure port is an integer and set to -1 if unknown
+                    port = -1
+                    if proxy.port:
+                        port = int(proxy.port)
```
Comment (poncovka) [inline pyanaconda/modules/subscription/subscription.py]: Please, use `port = int(proxy.port) if proxy.port else -1`.

File: pyanaconda/modules/subscription/subscription.py [hunk 2]
```diff
@@ -801,19 +806,28 @@ def set_server_proxy(self, hostname, port, username, password):
...
         # negative port number means no port has been set
         if port < 0:
             port_string = ""
         else:
             port_string = "{}".format(port)
+        # use "" when username is not known
+        username_string = ""
+        if username:
+            username_string = username
+        self._server_proxy_user = username_string
```
Comment (poncovka) [inline pyanaconda/modules/subscription/subscription.py]: Please, use `self._server_proxy_user = username or ""`.

File: pyanaconda/modules/subscription/subscription.py [hunk 3]
```diff
@@ -801,19 +806,28 @@ def set_server_proxy(self, hostname, port, username, password):
...
+        # use "" when password is not known
+        password_string = ""
+        if password:
+            password_string = password
+        self._server_proxy_password = password_string
```
Comment (poncovka) [inline pyanaconda/modules/subscription/subscription.py]: Please, use `self._server_proxy_password = password or ""`.

File: pyanaconda/modules/subscription/subscription.py (hunk 1 above)
Comment (M4rtinK) [reply pyanaconda/modules/subscription/subscription.py]: Hmm, yeah, this looks better. Changed.

File: pyanaconda/modules/subscription/subscription.py (hunk 2 above)
Comment (M4rtinK) [reply pyanaconda/modules/subscription/subscription.py]: Changed.

File: pyanaconda/modules/subscription/subscription.py (hunk 3 above)
Comment (M4rtinK) [reply pyanaconda/modules/subscription/subscription.py]: Changed.

**Pull request merged**